torch_dftd/functions/dftd3.py

```diff
@@ -283,21 +283,30 @@ def edisp(
             shift_abc = None if shift_abc is None else torch.cat([shift_abc, -shift_abc], dim=0)
         with torch.no_grad():
             # triplet_node_index, triplet_edge_index = calc_triplets_cycle(edge_index_abc, n_atoms, shift=shift_abc)
-            triplet_node_index, multiplicity, triplet_shift, batch_triplets = calc_triplets(
-                edge_index_abc, shift=shift_abc, dtype=pos.dtype, batch_edge=batch_edge_abc
+            # Type hinting
+            triplet_node_index: Tensor
+            multiplicity: Tensor
+            edge_jk: Tensor
+            batch_triplets: Optional[Tensor]
+            triplet_node_index, multiplicity, edge_jk, batch_triplets = calc_triplets(
+                edge_index_abc,
+                shift=shift_abc,
+                dtype=pos.dtype,
+                batch_edge=batch_edge_abc,
             )
             batch_triplets = None if batch_edge is None else batch_triplets
 
         # Apply `cnthr` cutoff threshold for r_kj
         idx_j, idx_k = triplet_node_index[:, 1], triplet_node_index[:, 2]
-        ts2 = triplet_shift[:, 2]
-        r_jk = calc_distances(pos, torch.stack([idx_j, idx_k], dim=0), cell, ts2, batch_triplets)
+        ts2 = None if shift_abc is None else shift_abc[edge_jk[:, 0]] - shift_abc[edge_jk[:, 1]]
```

Comment:
I think now you can change the variable name from `ts2` to other name. `shift_jk`?